### PR TITLE
k8s: CEP controller retries k8s version checks

### PR DESCRIPTION
**Pull request by raybejjani**
When requests for the k8s version are flaky, newly created CEP
controllers will not create the controller and never try again. The
check is now repeated in the controller until it succeeds, and the
result cached.

Fixes https://github.com/cilium/cilium/issues/6186

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6207)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

File: pkg/k8s/cep.go [hunk 1]
```diff
@@ -265,6 +256,26 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
...
+					k8sServerVer, err := GetServerVersion()
+					if err != nil {
+						scopedLog.WithError(err).Error("Unable to retrieve kubernetes server version")
+						return err
+					}
+					if !ciliumEPControllerLimit.Check(k8sServerVer) {
+						k8sVersionStop = true
+						scopedLog.WithFields(logrus.Fields{
+							"expected": k8sServerVer,
+							"found":    ciliumEPControllerLimit,
```
Comment (aanm) [inline pkg/k8s/cep.go]: The found/ expected are swapped.

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/k8s/cep.go (hunk 1 above)
Comment (raybejjani) [reply pkg/k8s/cep.go]: ah, so they are

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Comment by raybejjani** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by ianvernon** [comment]
Tests failed - related to CEP: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8020/

**Comment by ianvernon** [comment]
test-me-please

**Pull request merged**